### PR TITLE
fix(gitbook): parse all block types with a recursive renderer

### PR DESCRIPTION
**Pull request by Danelegend**
## Description

Customer report: pages with lists inside table cells were not searchable. Root cause: the GitBook parser was an allowlist of ~10 block types that silently dropped everything else, and its table handler only read single-paragraph cell fragments. An empty list item crashed the whole indexing run with an IndexError.

This replaces the parser with a recursive renderer:

- Blocks nest: lists in lists, code in quotes, blocks in fragments.
- Ordered (numbered) lists render; empty list items no longer crash the run.
- Table cells render through the full renderer, so lists inside cells work. Non-text columns (checkbox, number, select, content-ref) render by value type. Malformed records cannot crash the row sort, and unresolved fragment ids are dropped instead of leaking into indexed text.
- GitBook containers render: hints, tabs, expandable blocks, columns, embeds (URL), math (formula), image captions.
- Inline links render as `[text](url)` instead of vanishing.
- Unknown block types fall back to rendering children and fragments, so no content silently disappears again.

Output stays byte-compatible with the shapes the daily connector test asserts on.

Stack: 1/3 — followed by #14086 and #14087.

## How Has This Been Tested?

- New unit suite (`backend/tests/unit/onyx/connectors/gitbook/`) against a captured live API response exercising every block type, plus synthetic cases for list-in-table-cell, malformed records, empty list items, empty headings, and unknown blocks. 8/8 pass.
- Ran the connector against a live GitBook space: previously ~15 of 24 blocks on the fixture page were dropped; all content now indexes and is searchable in OpenSearch.

## Additional Options

- [ ] [Optional] Please cherry-pick this PR to the latest release version.
- [x] [Optional] Override Linear Check

---
## Summary by cubic
Replaces the GitBook allowlist parser with a recursive renderer to retain content and prevent crashes. Old: many blocks were dropped, inline links vanished, and empty/malformed list or table items could crash; New: nested blocks and full table cells render, links render as [text](url), unresolved fragment ids are ignored, and malformed table records and non-dict values are safely normalized.

- Review focus: `_render_block`, `_render_list`, `_render_table`, `_render_inline`, `_render_blocks`; `BlockType` and `InlineType` `StrEnum`s centralize handled types.
- Behavior: ordered lists render; blockquotes keep non-paragraph children; GitBook containers (hints, tabs, expandable, columns), embeds, math, and image captions now surface in text.
- Tables: cells render through the full renderer; non-text columns (checkbox, number, select, content-ref) render by value; malformed records/values no longer crash sort or rendering; unresolved fragment ids do not enter indexed text.
- Output: preserves shapes used by the daily connector test; downstream indexed text expands to include previously dropped content and link text.
- Tests: adds a unit suite with a captured API fixture and edge cases (lists in table cells, malformed records and record values, empty items, unknown blocks).

<sup>Written for commit e92cc5793463437a5a982123329243283ce67e93. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/onyx-dot-app/onyx/pull/14085?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: backend/onyx/connectors/gitbook/connector.py [hunk 1]
```diff
@@ -46,137 +46,309 @@ def get_page_content(self, space_id: str, page_id: str) -> dict[str, Any]:
...
+            for record_id, record in records.items()
+            if isinstance(record, dict)
+        ),
+        key=lambda kv: str(kv[1].get("orderIndex", "")),
+    )
+    for _record_id, record in sorted_records:
+        values = record.get("values") or {}
+        cells = [
+            _render_table_cell(
+                values.get(col), definition.get(col) or {}, fragments_by_id
```
[the remark this answers is not in the document]
Comment (Danelegend) [reply backend/onyx/connectors/gitbook/connector.py]: Fixed in f51d0c4: `values` is now normalized with an isinstance check before building cells, and the malformed-records unit test covers a truthy non-dict `values` field.

File: backend/onyx/connectors/gitbook/connector.py [hunk 2]
```diff
@@ -46,137 +46,309 @@ def get_page_content(self, space_id: str, page_id: str) -> dict[str, Any]:
...
+    sorted_records = sorted(
+        (
+            (record_id, record)
+            for record_id, record in records.items()
+            if isinstance(record, dict)
+        ),
+        key=lambda kv: str(kv[1].get("orderIndex", "")),
+    )
+    for _record_id, record in sorted_records:
+        values = record.get("values") or {}
```
[the remark this answers is not in the document]
Comment (Danelegend) [reply backend/onyx/connectors/gitbook/connector.py]: Fixed in f51d0c4: non-dict `values` are normalized to `{}` before building cells, with unit test coverage.

File: backend/onyx/connectors/gitbook/connector.py [hunk 3]
```diff
@@ -46,137 +46,310 @@ def get_page_content(self, space_id: str, page_id: str) -> dict[str, Any]:
...
+
+    if block_type == "blockquote":
+        return _quote_lines(_render_children(node, depth))
+
+    if block_type == "hint":
+        style = str(data.get("style", ""))
+        label = f"[!{style.upper()}]\n" if style else ""
+        return _quote_lines(label + _render_children(node, depth).strip("\n"))
+
+    if block_type == "tabs-item":
```
Comment (evan-onyx) [inline backend/onyx/connectors/gitbook/connector.py]: ideally there's an enum for this kind of thing but not a big deal.

**Review by evan-onyx — APPROVED** [review]

File: backend/onyx/connectors/gitbook/connector.py (hunk 3 above)
Comment (Danelegend) [reply backend/onyx/connectors/gitbook/connector.py]: Done — added `BlockType` and `InlineType` StrEnums and replaced the string literals in the renderer dispatch. The table column types ("text"/"select") stay as literals since they're a separate two-value vocabulary local to cell rendering.

**Comment by Danelegend** [comment]
@greptile
